### PR TITLE
a4a client plugin: initial infrastructure

### PR DESCRIPTION
**Pull request by jeherve**
## Proposed changes:

Introduce basic skeleton for the A4A client plugin.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* pfunGA-wi-p2

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Activate the plugin
* You should be redirected to its settings page.

### CODE REVIEW

File: projects/plugins/automattic-for-agencies-client/automattic-for-agencies-client.php
Comment (kraftbj) [inline projects/plugins/automattic-for-agencies-client/automattic-for-agencies-client.php]: I'm guessing it doesn't like the object being created without it being used itself somewhere. I'm not sure if this is a real problem we should attempt to address or if this is something to change the rules for. cc: @Automattic/jetpack-garage as the newly minted phan experts.

Comment (anomiex) [reply projects/plugins/automattic-for-agencies-client/automattic-for-agencies-client.php]: Phan's message does note that it might be being called for side effects of the constructor, which is very likely the case.

In this case it looks like we might do better to move what's in `__construct()` to a static `Automattic_For_Agencies_Client::init()` method instead, since nothing in there saves a reference to the new instance.

Comment (jeherve) [reply projects/plugins/automattic-for-agencies-client/automattic-for-agencies-client.php]: Done in ea021c469aa62527103f6f2418c70b092f040e4c

Comment (anomiex) [reply projects/plugins/automattic-for-agencies-client/automattic-for-agencies-client.php]: I note ea021c4 is a slight behavior change, in that all the `::init()` stuff is now done in the middle of loading plugins instead of waiting for the 'plugins_loaded' action. Does that matter?

Comment (jeherve) [reply projects/plugins/automattic-for-agencies-client/automattic-for-agencies-client.php]: I've tried to bring the `plugins_loaded` hook back in 7859ab333bcf1f711feed2ac08503cde334929c5, but had to revert due to problems with the plugin deactivation process. cc2939a2947ccabf2614bc881ca49ebaf6f1aea5

**Comment by rcanepa** [comment]
Hi @jeherve!

Thank you very much for creating the skeleton of this new plugin considering our product requirements (Jetpack Sync configuration, no My Jetpack, no entry on the sidebar, etc.). You saved us from a lot of back and forth by doing this for us.

I installed it in my local environment and tested a few things: the connection, integration with Manage, and A4A. Things work as expected, but we need to fix plenty of broken stuff on Calypso. Hopefully, some of them will benefit other standalone plugins that are in a similar situation.

Anyway, this is a great way to kick-start the plugin development, so thank you again.

**Comment by rcanepa** [comment]
@jeherve, is there any pending task before this PR can be merged?

**Comment by jeherve** [comment]
@rcanepa No, this PR just needs an approval, nothing else is needed for now.

**Review by rcanepa — DISMISSED** [review]
This first PR for the Automattic for Agencies Client plugin works as expected and fulfills some of the initial product requirements (no My Jetpack, no entry on the WPA sidebar, Jetpack Sync configuration, etc.), so I'm approving it to continue iterating on it.

**Comment by jeherve** [comment]
I'm afraid I'm going to need another approval, I had to rebase again.
Thank you!

**Review by rcanepa — APPROVED** [review]

**Pull request merged**
